.github/workflows/prerelease.yml

```diff
@@ -38,13 +45,59 @@ jobs:
          BOT_TOKEN="${ossbotaccess}" test_version="${nightly_version}" sdk_version_config="${GITHUB_WORKSPACE}/scripts/create_spec_repo/RC_firebase_sdk.textproto" local_sdk_repo_dir="${local_sdk_repo_dir}" podspec_repo_branch="${podspec_repo_branch}" scripts/release_testing_setup.sh prerelease_testing
+    - name: Clean spec repo
+      run: |
+         ossbotaccess=`cat oss-bot-access.txt`
+         git clone --quiet https://${ossbotaccess}@github.com/Firebase/SpecsTesting.git "${local_repo}"
+         cd "${local_repo}"
+         # Remove all unhided dirs, i.e. all podspec dir from the spec repo.
+         rm -Rf -- */
+         git add .
+         # commit without diff will throw an error. `git diff --exit-code` can avoid such error.
+         git diff --staged --exit-code || git commit -m "Empty spec repo."
+         git push
+    - name: Clean Artifacts
+      if: ${{ always() }}
+      run: |
+        rm -rf oss-bot-access.txt
+    - uses: actions/upload-artifact@v3
+      with:
+        name: firebase-ios-sdk
+        path: |
+          ${{ env.local_sdk_repo_dir }}/*.podspec
+          ${{ env.local_sdk_repo_dir }}/*.podspec.json
+  buildup_SpecsTesting_repo:
+    needs: specs_checking
+    # Don't run on private repo unless it is a PR.
+    if: (github.repository == 'Firebase/firebase-ios-sdk' && github.event_name == 'schedule') || github.event_name == 'workflow_dispatch'
+    runs-on: macos-12
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.specs_checking.outputs.matrix)}}
+    env:
+      bot_token_secret: ${{ secrets.GHASecretsGPGPassphrase1 }}
+      local_repo: specstesting
+      local_sdk_repo_dir: /tmp/test/firebase-ios-sdk
+      targeted_pod: ${{ matrix.podspec }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v3
+      with:
+        name: firebase-ios-sdk
+        path: ${{ env.local_sdk_repo_dir }}
+    - name: Get token
+      run: |
+         scripts/decrypt_gha_secret.sh scripts/gha-encrypted/oss-bot-access.txt.gpg \
+         oss-bot-access.txt "$bot_token_secret"
+         scripts/decrypt_gha_secret.sh scripts/gha-encrypted/bot-access.txt.gpg \
+         bot-access.txt "$bot_token_secret"
     - name: Update SpecsTesting repo
       run: |
         botaccess=`cat bot-access.txt`
         cd scripts/create_spec_repo/
         swift build
         pod repo add --silent "${local_repo}" https://"$botaccess"@github.com/Firebase/SpecsTesting.git
-        BOT_TOKEN="${botaccess}" .build/debug/spec-repo-builder --sdk-repo "${local_sdk_repo_dir}" --local-spec-repo-name "${local_repo}" --sdk-repo-name SpecsTesting --github-account Firebase --pod-sources 'https://${BOT_TOKEN}@github.com/Firebase/SpecsTesting' "https://github.com/firebase/SpecsDev.git" "https://github.com/firebase/SpecsStaging.git" "https://cdn.cocoapods.org/" --exclude-pods "FirebaseFirestoreTestingSupport" "FirebaseAuthTestingSupport" "FirebaseCombineSwift"
```

Comment:
Why don't we need ` --exclude-pods` any longer?

Reply:
The previous tool was searching targeted pods based on the file extensions, so we need to exclude some specs. In this update, we are using the [ManifestParser](https://github.com/firebase/firebase-ios-sdk/pull/9809/files#diff-72d9c3ffc2e64ae904528ac81111ab7de90c1e23e11a968000c97cb1d58135cdR53-R60)  to collect only podspecs matching specs in the Manifest file, so we don't need to manually exclude specs, which are not dependencies of any pods in the Manifest. 
One thing to notice is the `HeartbeatLoggingTestUtils` which depends on[ `FirebaseCoreInternal`](https://github.com/firebase/firebase-ios-sdk/blob/master/HeartbeatLoggingTestUtils.podspec#L37), but is also [a test dependency of `FirebaseCoreInternal`](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCoreInternal.podspec#L54), so a circular dependency warning of either `HeartbeatLoggingTestUtils`  or `FirebaseCoreInternal` could be found  in the log and the `HeartbeatLoggingTestUtils` will be pushed as well.